### PR TITLE
SONARIAC-550 Modify ArgInstruction implementation to use new KeyValuePair

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/TestUtils.java [hunk 1]
```diff
@@ -33,4 +36,13 @@ private TestUtils() {
...
   }
+
+  public static void assertKeyValuePair(NewKeyValuePair keyValuePair, String expectedKey, @Nullable String expectedValue) {
+    assertThat(argValue(keyValuePair.key())).isEqualTo(expectedKey);
+    if (expectedValue == null) {
+      assertThat(keyValuePair.value()).isNull();
+    } else {
+      assertThat(argValue(keyValuePair.value())).isEqualTo(expectedValue);
+    }
+  }
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/test/java/org/sonar/iac/docker/TestUtils.java:47]: Good idea, I start to understand the purpose of this class :D

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/ArgInstructionImplTest.java [hunk 2]
```diff
@@ -58,11 +59,9 @@ void argInstructionSimple() {
     assertThat(tree.children()).hasSize(2);
     assertThat(tree.keyValuePairs()).hasSize(1);
 
-    KeyValuePair keyValuePair = tree.keyValuePairs().get(0);
+    NewKeyValuePair keyValuePair = tree.keyValuePairs().get(0);
     assertThat(keyValuePair.getKind()).isEqualTo(DockerTree.Kind.KEY_VALUE_PAIR);
-    assertThat(keyValuePair.key().value()).isEqualTo("key1");
-    assertThat(keyValuePair.equals()).isNull();
-    assertThat(keyValuePair.value()).isNull();
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/ArgInstructionImplTest.java:65]: We globally lose the check on the `equals` operator.
Not necessarily a big miss.

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM ! Very nice idea for formatting the test !

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/ArgInstructionImplTest.java (hunk 2 above)
Comment (nils-werner-sonarsource) [reply iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/ArgInstructionImplTest.java:65]: Maybe we will have the rule to check for the equal sign. However, I think it is not important for now.

**Pull request merged**